- 可恢复错误
```Rust
enum Result<T,E> {
	Ok(T),
	Err(E),
}
```
- 不可恢复错误，一定会导致程序受到致命的打击而终止运行
```Rust
panic!
```
##### 一个例子
```Rust
use std::fs::File;

fn main() {
	let f = File::open("hello.txt");
	match f {
		Ok(file) => {
			println!("File open sucess");
		}
		Err(err) => {
			println!("File open failed");
		}
	}
	// 简化写法
	if let Ok(file) = f {
		println!("File open sucess");
	} else {
		println!("File open failed");
	}
}
```